app/modules/asset/service.server.ts

```diff
@@ -540,6 +540,46 @@ export async function deleteNote({
   });
 }
 
+export async function duplicateAsset({
+  asset,
+  userId,
+  amountOfDuplicates,
+}: {
+  asset: Prisma.AssetGetPayload<{ include: { custody: true } }>;
+  userId: string;
+  amountOfDuplicates: number;
+}) {
+  const duplicatedAssets = [];
+
+  for (let i = 0; i < amountOfDuplicates; i++) {
```

Comment:
This is fine, but we in our coding style we use `for...of` loops. Also for...of is async ready and I dont think a standard `for loop` is.